src/WordSearch/WordSearchSolver.ts

```diff
@@ -0,0 +1,98 @@
+import trie = require('trie-prefix-tree');
+import {Point} from './Point';
+import {WordSearchResult} from './WordSearchResult';
+
+export class WordSearchSolver {
+  // The grid of letters which makes up the wordsearch puzzle
+  private _matrix: string[][];
+  // Transpositions to apply when finding words
+  private _directions: number[][];
+  // Potential words to find in the letter grid
+  private _targets: any;  // tslint:disable-line
+
+  constructor(matrix: string[][]) {
+    this._matrix = matrix;
+
+    this._directions = [
+      // Cardinal directions
+      [0, 1], [-1, 0], [1, 0], [0, -1],
+      // Diagonal
+      [1, 1], [-1, -1], [1, -1], [-1, 1]
+    ];
+
+    this._targets = trie([]);
+  }
+
+  findWords(words: string[]): WordSearchResult[] {
+    for (const word of words) {
+      if (word !== '') {

```

Comment:
What about null, undefined, numbers, etc?  Likely we should throw in cases where an invalid input is passed, right?

Reply:
I'll add a check for null, but I'd like to keep this as open as possible.  I think as soon as I restrict it to alphabetic characters, the next puzzle is going to be wordsearch of just numbers :P

Reply:
Do we really want to silently ignore empty strings? What about a string containing only spaces? I would argue that you want to trim first and throw an error in the empty case.

Reply:
I also was referring to the literal type number, it seems like this code assumes a string so if something isn't a string I don't know what the expectation is.